### DISCUSSION TITLE
[RFC] Remove Angular modules

### DISCUSSION THREAD

**Discussion by LayZeeDK** [ngworker/lumberjack#168] [RFC]
Angular modules are no longer necessary. We already have `provideLumberjack`, `provideLumberjackConsoleDriver`, and `provideLumberjackHttpDriver` but we still have Angular module alternatives.

## Pros

- Decreased bundle size
- Smaller public API surface
- Exposing a function is less confusing than exposing both an Angular module and a function that offer the same features
- Provider features are tree-shakable on a granular level, Angular module-based APIs are not

## Cons

- Breaking change
   - Mitigated by deprecating the Angular modules during a grace period and writing a migration guide

**Comment by NachoVazquez** [comment]
I'm in favor. And one major version warning before deprecation should be enough. For example, we should start deprecating the Modules in version 16.

**Comment by rainerhahnekamp** [comment]
Definitely YES

**Comment by santoshyadavdev** [comment]
Yup let's drop them, we can provide a migration guide as well. To make migration to standalone smooth.
